### PR TITLE
Implementing OpenApi 3 (converted from Swagger2)

### PR DESCRIPTION
**Pull request by leandro-deveikis**
**What this PR does / why we need it**:
This PR adds a new script to take the Swagger 2 specs (already generated) and converts them to OpenAPI 3
It also adds a new endpoint to load swagger UI with the new specs

**Special notes for your reviewer**:
* As for now, we don't add a new feature toggle, the new endpoint, as it didn't seem necessary. If the new specs are not generated, but the endpoint is accessed, it would only show an error saying that the spec file was not found. 
* The new endpoint is served at `/openapi3`

### CODE REVIEW

File: public/openapi3.json [hunk 1]
```diff
@@ -0,0 +1,9983 @@
+{
```
Comment (papagian) [inline public/openapi3.json:1]: We have to let prettier ignore the `public/openapi3.json` same as it's done [here](53170)

File: scripts/openapi3/openapi3conv.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/getkin/kin-openapi/openapi2"
+	"github.com/getkin/kin-openapi/openapi2conv"
+	"io/ioutil"
+	"os"
+)
+
+// main This simple script will take the swagger 2 specs generated by grafana and convert them into openapi 3
```
Comment (papagian) [inline scripts/openapi3/openapi3conv.go:12]: I think it takes only one specification as input

**Review by papagian — COMMENTED** [review]
LGTM but when I try out a request from the Swagger UI it fails because [servers](https://swagger.io/docs/specification/api-host-and-base-path/) that replaces the` host`, [basePath](https://github.com/grafana/grafana/blob/main/public/api-merged.json#L27) and [schemes](https://github.com/grafana/grafana/blob/6b8316d510f771629325e0afba0de33c508b0ce9/public/api-merged.json#L8) keywords used in OpenAPI 2.0, is missing from the generate file ():
`http://localhost:3000/users?perpage=1000&page=1`
instead of
`http://localhost:3000/api/users?perpage=1000&page=1`

Adding the following works for me. However, I'm not sure how in that way we can have the the URL get resolved against the server that hosts the given OpenAPI definition (and not the localhost), following [this](https://swagger.io/docs/specification/api-host-and-base-path/#relative-urls) pattern :
```
	"servers": [
		{
			"url": "{protocol}://localhost:3000/api",
			"variables": {
				"protocol": {
					"enum": [
						"http",
						"https"
					],
					"default": [
						"https"
					]
				}
			}
		}
	],
```

I'm not sure it that's blocking though; we can improve it in a follow-up

File: Makefile [hunk 3]
```diff
@@ -56,6 +56,10 @@ validate-api-spec: $(MERGED_SPEC_TARGET) $(SWAGGER) ## Validate API spec
 clean-api-spec:
 	rm $(SPEC_TARGET) $(MERGED_SPEC_TARGET)
```
[the remark this answers is not in the document]
Comment (leandro-deveikis) [reply Makefile]: Fixed in aefef610fa592330337125ac5e1ff9ccaebbb2ed

File: public/openapi3.json (hunk 1 above)
Comment (leandro-deveikis) [reply public/openapi3.json]: Fixed in aefef610fa592330337125ac5e1ff9ccaebbb2ed

**Comment by leandro-deveikis** [comment]
> LGTM but when I try out a request from the Swagger UI it fails because [servers](https://swagger.io/docs/specification/api-host-and-base-path/) that replaces the` host`, [basePath](https://github.com/grafana/grafana/blob/main/public/api-merged.json#L27) and [schemes](https://github.com/grafana/grafana/blob/6b8316d510f771629325e0afba0de33c508b0ce9/public/api-merged.json#L8) keywords used in OpenAPI 2.0, is missing from the generate file (): `http://localhost:3000/users?perpage=1000&page=1` instead of `http://localhost:3000/api/users?perpage=1000&page=1`
> 

Thanks for the feedback @papagian 
To be honest, I thought I  tested it but seems that it was not working correctly

Looking into this, the problem seems to be the difference in both specs. [Swagger2](https://swagger.io/docs/specification/2-0/api-host-and-base-path/) has a complete struct with different attributes, however, as you mentioned, [OpenAPI3](https://swagger.io/docs/specification/api-host-and-base-path/) has an array of server urls in which we could use some variables (in case we want to support http and https for instance). Given this difference, if there is no host provided on the Swagger2 spec (which we dont have), the converter will not create any server url. [Check here](https://github.com/getkin/kin-openapi/blob/8287d3697f36b53c0aca5926bab441b0de465ae7/openapi2conv/openapi2_conv.go#L28)
As a workaround, in this commit 22ee3b69e7561ed7d4a42b8700325dbdba2d2ee2 I manually added the "/api" as a server url. This would mean that it would take it as a prefix and use the host where the swagger iu is being served. 
With this, I could test some calls successfully. IMO this is enough for now

**Review by papagian — APPROVED** [review]
great work!

**Pull request merged**